compliance/azure/instances_without_fnm_agent/Azure_Instances_not_running_FlexNet_Inventory_Agent.pt

```diff
@@ -0,0 +1,194 @@
+    path join(["/subscriptions/", $param_azure_sub, "/providers/Microsoft.Compute/virtualMachines"])
+    query "api-version","2018-06-01"
+    header "User-Agent", "RS Policies"
+  end
+  result do
+    encoding "json"
+    collect jmes_path(response, "value[*]") do
+        field "id", jmes_path(col_item,"id")
+        field "name", jmes_path(col_item,"name")
+        field "resourcename", get(4, split(jmes_path(col_item, "id"),"/"))
+        field "location", jmes_path(col_item,"location")
+        field "vmSize", jmes_path(col_item,"properties.hardwareProfile.vmSize")
+        field "osType", jmes_path(col_item,"properties.storageProfile.osDisk.osType")
+        field "tags", jmes_path(col_item,"tags")
+    end
+  end
+end
+
+datasource 'ds_fnms_report' do
+  request do
+    auth $fnms_api_key_auth
+    host "www.flexnetmanager.com"
+    path '/ManageSoftServices/ComplianceAPIService/ComplianceAPIService.asmx'
+    verb 'POST'
+    scheme 'https'
+    header "Content-Type",'text/xml;charset=utf-8'
+    body join(['<?xml version="1.0" encoding="utf-8"?>
+    <soap:Envelope xmlns:soap="http://www.w3.org/2003/05/soap-envelope" xmlns:tem="http://tempuri.org/">
+      <soap:Body>
+        <tem:GetCustomView>
+          <tem:customViewID>',$param_report_id,'</tem:customViewID>
+          <tem:rowLimit>1000</tem:rowLimit>
+        </tem:GetCustomView>
+      </soap:Body>
+    </soap:Envelope>'])
+  end
+  result do
+    encoding "xml"
+    collect xpath(response,"//NewDataSet/SearchResults") do
+      field "computerName", xpath(col_item, "ComputerName")
+    end
+  end
+end
+
+datasource "ds_formatted_instances", type: "javascript" do
+  run_script $js_format_instances, $ds_list_virtual_machines, $ds_fnms_report, $param_exclusion_tag_key
+end
+
+###############################################################################
+# Scripts
+###############################################################################
+
+#Formatter script for the Instance data
+script "js_format_instances", type: "javascript" do
+  parameters "azure_VMs","ds_fnms_report", "param_exclusion_tag_key"
+  result "result"
+  code <<-EOS
+    var result = [];
+  fnms_instance_name=[];
+    _.each(ds_fnms_report, function(instance){
+      fnms_instance_name.push(instance["computerName"])
+    })
+
+    _.each(azure_VMs, function(vm){
+      if((!_.has(vm.tags, param_exclusion_tag_key)) && fnms_instance_name.indexOf(vm["name"]) === -1){
+        result.push(vm)
+      }
+    })
+  EOS
+end
+
+###############################################################################
+# Policy
+###############################################################################
+
+policy "policy_long_running_instances" do
+  validate $ds_formatted_instances do
+    summary_template "{{ rs_project_name }} (Account ID: {{ rs_project_id }}): {{len data }} is missing in FlexNet Manager."
+    detail_template <<-EOS
+| VM Name | Resource Name | location | vmSize | OS Type | tags |
```

Comment:
First letter should be upper case for location and tags, vmSize should be VM Size.

Reply:
Made the required changes. Thanks.